### PR TITLE
Implement some cheats and showing cheat activated message

### PR DESCRIPTION
**Pull request by NFSMONSTR**

### CODE REVIEW

File: rwgame/RWGame.cpp [hunk 1]
```diff
@@ -212,31 +216,40 @@ void RWGame::handleCheatInput(char symbol) {
...
+                                           20, //shotgun
+                                           5,  //grenade
+                                           5,  //molotov
+                                           5,  //rocket launcher
+                                           20, //flamethrower
+                                           200,//ak47
+                                           200,//m16
+                                           5   //sniper rifle
+                                           };
+            for (unsigned long i = 0; i < ammo.size(); i++)
```
Comment (husho) [inline rwgame/RWGame.cpp]: better to use ```std::size_t```

Comment (husho) [reply rwgame/RWGame.cpp]: or even ```size_type```

Comment (NFSMONSTR) [reply rwgame/RWGame.cpp]: Thank you

Comment (NFSMONSTR) [reply rwgame/RWGame.cpp]: It seems like that using `size_type` will cause warning about comparison of integer expressions of different signedness. So I'll think it better to use something like `std::vector<size_type>::size_type`

Comment (husho) [reply rwgame/RWGame.cpp]: Yeah, that's what I meant, it should be  ```std::vector<int>::size_type```

File: rwgame/RWGame.cpp [hunk 2]
```diff
@@ -212,31 +216,40 @@ void RWGame::handleCheatInput(char symbol) {
...
-            // @todo ShowHelpMessage("CHEAT4"); // III / VC: Inputting armor
-            // cheat.
+            showHelpMessage("CHEAT4"); // III / VC: Inputting armor cheat.
         });
 
         checkForCheat("GUNSGUNSGUNS", [&] {
-            // @todo give player weapons
-            // @todo ShowHelpMessage("CHEAT2"); // III / VC: Inputting weapon
-            // cheats.
+            const std::vector<int> ammo = {1,  //baseball bat
```
Comment (ShFil119) [inline rwgame/RWGame.cpp]: It can even be a std::array.

File: rwgame/RWGame.cpp [hunk 3]
```diff
@@ -245,51 +258,50 @@ void RWGame::handleCheatInput(char symbol) {
...
+         uint16_t vehicleModel = 110;  // @todo Which cars are spawned?!
 #else
-        // uint16_t vehicleModel = 122;
+         uint16_t vehicleModel = 122;
 #endif
-            // @todo Spawn rhino
-            // @todo ShowHelpMessage("CHEAT1"); // III / VC: Inputting most
-            // cheats.
+            auto ch = getPlayer()->getCharacter();// @todo Change spawn place to be more like in original game
+            auto playerRot = ch->getRotation();
```
Comment (ShFil119) [inline rwgame/RWGame.cpp]: Can this and below vars be const?

File: rwgame/RWGame.cpp [hunk 4]
```diff
@@ -160,6 +160,11 @@ void RWGame::startScript(const std::string& name) {
     }
 }
 
+void RWGame::showHelpMessage(const GameStringKey& id) {
```
Comment (husho) [inline rwgame/RWGame.cpp]: This should be a member of GameState

**Comment by husho** [comment]
Needs rebase

**Comment by NFSMONSTR** [comment]
I will just recreate it normally and updated

**Comment by JayFoxRox** [comment]
This was still fixable - in the future, avoid closing PRs unless there is a fundamental design issue or it's not necessary anymore.
By closing the PR you broke GitHubs internal mechanism, so you can't re-open this one without actually causing issues.

**Comment by NFSMONSTR** [comment]
Sorry, I still weak with GitHub and Git

**Pull request closed without merging**